documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,232 @@ Control characters are not allowed in a specification.
+
+**Why the rule is important**: Not adhering to the rule violates how the x-ms-mutability extension works. A property cannot be `readonly: true` and yet allow `x-ms-mutability` as `["create", "update"]`. 
+
+**How to fix the violation**: Based on the value of the `readOnly` property, assign appropriate `x-ms-mutability` options. 
+
+**Bad Example**:
+```
+  "prop0":{
+    "type": "string",
+    "readOnly":true,
+    "x-ms-mutability": ["read", "update"]
+  }
+```
+**Good Example**:
+```
+  "prop0":{
+    "type": "string",
+    "readOnly": false,
+    "x-ms-mutability": ["read", "update"]
+  }
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2058" />R2058 XmsPathsMustOverloadPaths 
+**Output Message**: Paths in x-ms-paths must overload a normal path in the paths section, i.e. a path in the x-ms-paths must either be same as a path in the paths section or a path in the paths sections followed by additional parameters. 
+
+**Description**: The `x-ms-paths` extension allows us to overload an existing path based on path parameters. We cannot specify an `x-ms-paths` without a path that already exists in the `paths` section. For more details about this extension please refere [here](https://github.com/Azure/azure-rest-api-specs/blob/dce4da0d748565efd2ab97a43d0683c2979a974a/documentation/swagger-extensions.md#x-ms-paths).
+
+**Why the rule is important**: The `x-ms-paths` overload an existing path only, not adhering to this rule would violate the applicability of the extension itself.
+
+**How to fix the violation**: Ensure that the `x-ms-paths` is overloading an existing url path in the `paths` section.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2001" />S2001 AvoidNestedProperties
+**Output Message**: Consider using x-ms-client-flatten to provide a better end user experience
+
+**Description**: Nested properties can result into bad user experience especially when creating request objects. `x-ms-client-flatten` flattens the model properties so that the users can analyze and set the properties much more easily.
+
+**Why the rule is important**: Overly nested properties (especially required ones) can result into a bad user experience.
+
+**How to fix the violation**: Either eliminate nesting or use the `x-ms-client-flatten` property for a better user experience. More details about the extension can be found [here](https://github.com/Azure/azure-rest-api-specs/blob/dce4da0d748565efd2ab97a43d0683c2979a974a/documentation/swagger-extensions.md#x-ms-client-flatten).
```

Comment:
Do we need to provide information about "Falttenning Threshold" that is one of the autorest parameter ?